### PR TITLE
Fixed removing the password expiration date

### PR DESCRIPTION
**Pull request by mvidner**
Merging #153 to master

plus some minimal refactoring of the affected method, which had 1900 lines. And the obviously correct refactoring obviously introduced a stupid bug which I found with manual testing.

Plus Coveralls and Yardoc.

### CODE REVIEW

File: src/include/users/dialogs.rb [hunk 1]
```diff
@@ -155,6 +155,145 @@ def AskForOldPassword
...
+            ),
+            VSpacing(0.2)
+          ),
+          HSpacing(1)
+        )
+      )
+      ui = UI.UserInput
+      retmap = { "retval" => ui == :yes }
+      if ui == :yes
+        Ops.set(retmap, "chown_home", UI.QueryWidget(Id(:chown_home), :Value))
```
Comment (jreidinger) [inline src/include/users/dialogs.rb]: set is not needed here as above it creates solid hash.

File: src/include/users/dialogs.rb [hunk 2]
```diff
@@ -155,6 +155,145 @@ def AskForOldPassword
...
+              PushButton(Id(:yes), Opt(:default), Label.YesButton),
+              PushButton(Id(:no), Label.NoButton)
+            ),
+            VSpacing(0.2)
+          ),
+          HSpacing(1)
+        )
+      )
+      ui = UI.UserInput
+      retmap = { "retval" => ui == :yes }
```
Comment (jreidinger) [inline src/include/users/dialogs.rb]: two times identical comparison to `:yes`. Maybe own variable?

File: src/include/users/dialogs.rb [hunk 3]
```diff
@@ -155,6 +155,145 @@ def AskForOldPassword
...
+      min = GetInt(Ops.get(user, "shadowMin"), -1)
+      warn = GetInt(Ops.get(user, "shadowWarning"), -1)
+
+      if last_change != "0"
+        last_change_label = format_days_after_epoch(last_change, "%x")
+      else
+        # label (date of last password change)
+        last_change_label = _("Never")
+      end
+      if expires != "0" && expires != "-1" && expires != ""
```
Comment (jreidinger) [inline src/include/users/dialogs.rb]: `if !["0", "-1", ""].include?(expires)`

**Comment by jreidinger** [comment]
in general looks good code wise, but what I miss pretty heavily is any sort of test coverage.

**Comment by mvidner** [comment]
@jreidinger thanks for the comments; updated

**Pull request merged**